cumulus/parachains/runtimes/bridge-hubs/bridge-hub-rococo/tests/snowbridge.rs

```diff
@@ -206,3 +212,40 @@ fn construct_and_apply_extrinsic(
 	let r = Executive::apply_extrinsic(xt);
 	r.unwrap()
 }
+
+// The event log from https://bridgehub-rococo.stg.subscan.io/extrinsic/2583529-2 to reproduce an issue
+// run this test with ```cargo test --release -p bridge-hub-rococo-runtime --test snowbridge
+// send_token_to_foreign_chain_with_invalid_dest_fee -- --nocapture```
+#[test]
+#[cfg(not(debug_assertions))]
+pub fn send_token_to_foreign_chain_with_invalid_dest_fee() {
```

Comment:
Maybe I am missing the point here, but what purpose does this test serve? Is it just to replicate what happen in Rococo?

Reply:
Yeah, this PR is more of reproduce than fix. No need to merge if [1148](https://github.com/Snowfork/snowbridge/pull/1148) got accepted.

Reply:
Yeah, agree. Why would a zero fee cause an error like SendError::ExceedsMaxMessageSize? Do we know the source of the error?

Reply:
Confirmed, zero Fungible asset is not allowed which fail the check [here](https://github.com/Snowfork/polkadot-sdk/blob/66505a77858d5091438f5362f53cf5d317274c30/polkadot/xcm/src/v4/asset.rs#L285)

Though the error message could be improved(`ExceedsMaxMessageSize` a bit confusing) but not the scope of this PR. So https://github.com/Snowfork/snowbridge/pull/1148 should be the right fix.

Reply:
Nice find.

Do we want to consider changing the generated xcm to not BuyExecution with zero fees?
```diff
diff --git a/bridges/snowbridge/primitives/router/src/inbound/mod.rs b/bridges/snowbridge/primitives/router/src/inbound/mod.rs
index c20554c6d1..bff3b15f1a 100644
--- a/bridges/snowbridge/primitives/router/src/inbound/mod.rs
+++ b/bridges/snowbridge/primitives/router/src/inbound/mod.rs
@@ -254,16 +254,22 @@ where
 
 		match dest_para_id {
 			Some(dest_para_id) => {
-				let dest_para_fee_asset: Asset = (Location::parent(), dest_para_fee).into();
+				let mut assets = vec![asset.clone()];
+				let mut execution = UnpaidExecution { weight_limit: Unlimited, check_origin: None };
+				if dest_para_fee > 0 {
+					let dest_para_fee_asset: Asset = (Location::parent(), dest_para_fee).into();
+					assets.push(dest_para_fee_asset.clone());
+					execution = UnpaidExecution { weight_limit: Unlimited, check_origin: None };
+				}
 
 				instructions.extend(vec![
 					// Perform a deposit reserve to send to destination chain.
 					DepositReserveAsset {
-						assets: Definite(vec![dest_para_fee_asset.clone(), asset.clone()].into()),
+						assets: Definite(assets),
 						dest: Location::new(1, [Parachain(dest_para_id)]),
 						xcm: vec![
 							// Buy execution on target.
-							BuyExecution { fees: dest_para_fee_asset, weight_limit: Unlimited },
+							execution,
 							// Deposit asset to beneficiary.
 							DepositAsset { assets: Definite(asset.into()), beneficiary },
 						]

```

Reply:
I prefer to not introduce it, IMO usually parachains won't allow zero fees so more of an input error here. Even if they allow that it also requires some custom xcm config to `AllowUnpaidExecutionFrom` ony from AssetHub(not from other chain) which we not test yet.